src/openms_gui/source/VISUAL/LayerData.cpp

```diff
@@ -217,4 +217,80 @@ namespace OpenMS
     if (annotations_changed) { hit.setPeakAnnotations(fas); }
   }
 
+  void LayerData::removePeakAnnotationsFromPeptideHit(std::vector<Annotation1DItem*> anno_items)
+  {
+    // LayerData & current_layer = widget_1D->canvas()->getCurrentLayer();
+    int spectrum_index = getCurrentSpectrumIndex();
+
+    // Return if no valid peak layer attached
+    if (getPeakData()->size() == 0 || type != LayerData::DT_PEAK) { return; }
+
+    MSSpectrum & spectrum = (*getPeakData())[spectrum_index];
+    int ms_level = spectrum.getMSLevel();
+
+    if (ms_level != 2)
+    {
+      return;
+    }
+    else
+    {
+      // store user fragment annotations
+      vector<PeptideIdentification>& pep_ids = spectrum.getPeptideIdentifications();
+
+      // no ID selected
+      if (peptide_id_index == -1 || peptide_hit_index == -1)
+      {
+        return;
+      }
+
+      if (pep_ids.empty())
```

Comment:
maybe just
if (...) { return; }
and get rid of the else block (might be easier to read)
Same in line 254.